File: GalileoX_CAAD003X_Professional_Android_App_Development/DaggerDemo/app/src/main/java/com/example/robert/daggerdemo/di/ActivityModule.java
package com.example.robert.daggerdemo.di;

import android.app.Activity;
import android.content.Context;

import dagger.Module;
import dagger.Provides;

/**
 * Created by robert on 23.7.2017.
 */

@Module
public class ActivityModule {

    private Activity mActivity;

    public ActivityModule(Activity activity) {
        this.mActivity = activity;
    }

    @Provides
    @ActivityContext
    Context provideContext() {
        return mActivity;
    }

    @Provides
    Activity provideActivity() {
        return mActivity;
    }
}

File: Android/GalileoX_CAAD003X_Professional_Android_App_Development/PhotoLocation/app/src/main/java/com/example/robert/photolocation/di/ActivityModule.kt
package com.example.robert.photolocation.di

import com.example.robert.photolocation.view.MainActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * Created by robert on 1.10.2017.
 */
@Module
abstract class ActivityModule {
    @ContributesAndroidInjector
    abstract fun bindMainActivity(): MainActivity
}